src/ros_filter.cpp

```diff
@@ -1541,6 +1584,14 @@ namespace RobotLocalization
 
     while (ros::ok())
     {
+      // Each second we remove old states and measurements from the queue.
+      static int flush_counter = 0;
+      flush_counter++;
+      if (flush_counter > frequency_)
```

Comment:
Correct me if I'm wrong (it's early here!), but this appears to only save us from calling this method for the first `frequency` times through this loop, which means we're probably avoiding on the order of tens to hundreds of function calls total. The remove method will look for filter states older than the time given, and won't find any, correct? If I'm correct, I think we're safe to remove the if check. If not, can we move the flush_counter out of the loop and make it not static?